pkg/clustermanager/cluster_manager.go

```diff
@@ -177,7 +233,42 @@ func (c *ClusterManager) DeleteCluster(ctx context.Context, managementCluster, c
 }
 
 func (c *ClusterManager) UpgradeCluster(ctx context.Context, managementCluster, workloadCluster *types.Cluster, clusterSpec *cluster.Spec, provider providers.Provider) error {
-	if err := c.applyCluster(ctx, managementCluster, workloadCluster, clusterSpec, provider, true, types.WithNodeRef(), types.WithNodeHealthy()); err != nil {
+	cpContent, mdContent, err := provider.GenerateClusterApiSpecForUpgrade(ctx, managementCluster, workloadCluster, clusterSpec)
+	if err != nil {
+		return fmt.Errorf("error generating capi spec: %v", err)
+	}
+
+	if err = c.writeCapiSpecFile(clusterSpec.ObjectMeta.Name, templater.AppendYamlResources(cpContent, mdContent)); err != nil {
+		return err
+	}
+
+	err = c.Retrier.Retry(
+		func() error {
+			return c.clusterClient.ApplyKubeSpecFromBytesWithNamespace(ctx, managementCluster, cpContent, constants.EksaSystemNamespace)
```

Comment:
since the mdContent is appended to cpContent, will the kubectl apply command apply the updated spec for cp and md at the same time? should we first wait for controlplane upgrade to go through and then apply mdContent separately?

Reply:
yea that's what i am doing. the mdContent is only appended to cp for writing the spec file to local. we still apply cpContent first, then mdContent.

Reply:
ah gotcha, I thought append yaml resources on line 241 was appending mdContent to cpContent so cpContent is getting updated, but that's not the case